website/docs/r/monitor_data_collection_rule_association.html.markdown

```diff
@@ -0,0 +1,154 @@
+  name                = "machine"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  size                = "Standard_B1ls"
+  admin_username      = "adminuser"
+  network_interface_ids = [
+    azurerm_network_interface.example.id,
+  ]
+
+  admin_password = "example-Password@7890"
+
+  disable_password_authentication = false
+
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "UbuntuServer"
+    sku       = "16.04-LTS"
+    version   = "latest"
+  }
+}
+
+resource "azurerm_monitor_data_collection_rule" "example" {
+  name                = "example-dcr"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+  destinations {
+    azure_monitor_metrics {
+      name = "example-destination-metrics"
+    }
+  }
+  data_flow {
+    streams      = ["Microsoft-InsightsMetrics"]
+    destinations = ["example-destination-metrics"]
+  }
+}
+
+resource "azurerm_monitor_data_collection_endpoint" "test" {
+  name                = "example-dce"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = azurerm_resource_group.example.location
+}
+
+# associate to a Data Collection Rule
+resource "azurerm_data_collection_rule_association" "example1" {
+  name                    = "example1-dcra"
+  target_resource_id      = azurerm_linux_virtual_machine.example.id
+  data_collection_rule_id = azurerm_monitor_data_collection_rule.example.id
+  description             = "example"
+}
+
+# associate to a Data Collection Endpoint
+resource "azurerm_data_collection_rule_association" "example1" {
+  name                        = "configurationAccessEndpoint"
+  target_resource_id          = azurerm_linux_virtual_machine.example.id
+  data_collection_endpoint_id = azurerm_monitor_data_collection_endpoint.example.id
+  description                 = "example"
+}
+
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name which should be used for this Data Collection Rule. Changing this forces a new Data Collection Rule to be created.
+
+* `target_resource_id` - (Required) The ID of the Azure Resource which to associate to a Data Collection Rule or a Data Collection Endpoint. Changing this forces a new resource to be created.
+
+---
+
+* `data_collection_endpoint_id` - (Optional) The ID of the Data Collection Endpoint which will be associated to the target resource.
+
+* `data_collection_rule_id` - (Optional) The ID of the Data Collection Rule which will be associated to the target resource.
+ 
+-> **NOTE** Exactly one of `data_collection_endpoint_id` and `data_collection_rule_id` blocks must be specified.
+
+-> **NOTE** An association of Data Collection Endpoint must be named 'configurationAccessEndpoint'. 
```

Comment:
this should be below the name property?

Reply:
I think maybe this description is better
```suggestion
-> **NOTE** If `data_collection_endpoint_id` is specified, this data collection rule association must be named 'configurationAccessEndpoint'. 
```